provider/on_call_schedules_data_test.go

```diff
@@ -0,0 +1,76 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/stretchr/testify/suite"
+)
+
+/** Suite *************************************************************************************************************/
+type testOnCallSchedulesDataSuite struct {
+	suite.Suite
+}
+
+func TestOnCallSchedulesData(t *testing.T) {
+	suite.Run(t, new(testOnCallSchedulesDataSuite))
+}
+
+func (s *testOnCallSchedulesDataSuite) terraform(rName string) string {
+	return fmt.Sprintf(`
+resource "firehydrant_team" "test_on_call_schedules_data_team" {
+	name = "test-team-%s"
+}
+
+resource "firehydrant_on_call_schedule" "test_on_call_schedule_data_1" {
+  name        = "test-on-call-schedule-%s"
+  description = "test-description"
+  team_id     = firehydrant_team.test_on_call_schedules_data_team.id
+  time_zone   = "America/Los_Angeles"
+
+  strategy {
+	type         = "weekly"
+	handoff_time = "10:00:00"
+	handoff_day  = "thursday"
+  }
+
+  restrictions {
+    start_day  = "monday"
+    start_time = "14:00:00"
+    end_day    = "friday"
+    end_time   = "17:00:00"
+  }
+}
+
+data "firehydrant_on_call_schedules" "test_on_call_schedules_data" {
+	team_id = firehydrant_team.test_on_call_schedules_data_team.id
+	query = "test-on-call-schedule"
+}`, rName, rName)
+}
+
+func (s *testOnCallSchedulesDataSuite) testResource(steps ...resource.TestStep) {
+	resource.Test(s.T(), resource.TestCase{
+		PreCheck:          func() { testFireHydrantIsSetup(s.T()) },
+		ProviderFactories: defaultProviderFactories(),
+		Steps:             steps,
+	})
+}
+
+/** Tests *************************************************************************************************************/
+
+func (s *testOnCallSchedulesDataSuite) TestMatches() {
+	rName := acctest.RandStringFromCharSet(20, acctest.CharSetAlphaNum)
+	s.testResource(resource.TestStep{
+		Config: s.terraform(rName),
+		Check: resource.ComposeAggregateTestCheckFunc(
```

Comment:
For both `schedule` and `schedules`, docs suggest that this should also contain `slack_user_group_id` and `color`.  We should either be testing for those attributes or removing them from the docs as unsupported.

Reply:
I'd say to do the simple thing for expediency's sake.  I'm not so concerned if it's here or not, just that the tests accurately reflect the docs so someone else doesn't try to use this and it doesn't meet expectations.

Reply:
@AlexisJasso I went ahead and added the `slack_user_group_id` attribute (as well as now testing for `color` too), as it was trivial to implement. 🙇  ♻️